scripts/generate_error_code.hs

```diff
@@ -0,0 +1,257 @@
+      }
+  , ErrorItem
+      { eiLabel = "position_not_exist_err"
+      , eiCode = 101
+      , eiDesc = "Thrown by `collect_fees`: Position does not exist."
+      }
+  , ErrorItem
+      { eiLabel = "log_out_of_bounds_err"
+      , eiCode = 102
+      , eiDesc = "Thrown by `floor_log_half_bps`: Log out of bounds."
+      }
+  , ErrorItem
+      { eiLabel = "end_ladder_reached_err"
+      , eiCode = 103
+      , eiDesc = "Thrown by `half_bps_pow_rec`: Should not reach end of ladder."
+      }
+  , ErrorItem
+      { eiLabel = "past_deadline_err"
+      , eiCode = 104
+      , eiDesc = "Thrown by `x_to_y` and `y_to_x`: Past deadline."
+      }
+  , ErrorItem
+      { eiLabel = "smaller_than_min_asset_err"
+      , eiCode = 105
+      , eiDesc = "Threshold on amount of bought tokens violated: `dx` received < `min_dx` or `dy` received < `min_dy`"
+      }
+  , ErrorItem
+      { eiLabel = "asset_transfer_invalid_entrypoints_err"
+      , eiCode = 106
+      , eiDesc = "There is no appropriate transfer entrypoint in the provided token contract to transfer asset `x` or `y`."
+      }
+  , ErrorItem
+      { eiLabel = "tick_not_exist_err"
+      , eiCode = 107
+      , eiDesc = "User provided tick is not initialized."
+      }
+  ]
+
+internalErrors :: [ErrorItem]
+internalErrors =
+  [ ErrorItem
+      { eiLabel = "internal_tick_not_exist_err"
+      , eiCode = 200
+      , eiDesc = "Tick is not initialized."
+      }
+  , ErrorItem
+      { eiLabel = "internal_epoch_bigger_than_now_err"
+      , eiCode = 201
+      , eiDesc = "Time now is smaller than epoch time."
+      }
+  , ErrorItem
+      { eiLabel = "internal_fee_more_than_100_percent_err"
+      , eiCode = 202
+      , eiDesc = "The `const_fee_bps` is initialized to be higher than 10000 (100%)."
+      }
+  , ErrorItem
+      { eiLabel = "internal_203"
+      , eiCode = 203
+      , eiDesc = "Throw when `(p.s.sqrt_price - sqrt_price_new)` is not nat"
+      }
+  , ErrorItem
+      { eiLabel = "internal_204"
+      , eiCode = 204
+      , eiDesc = "Throw when `(sqrt_price_new - p.s.sqrt_price)` is not nat."
+      }
+  , ErrorItem
+      { eiLabel = "flip_fee_growth_outside_err"
+      , eiCode = 205
+      , eiDesc = "Flip for `fee_growth_outside` failed. (This is an invariant of the contract)"
+      }
+  , ErrorItem
+      { eiLabel = "internal_206"
+      , eiCode = 206
+      , eiDesc = "Throw when `(p.dx - dx_consummed)` is not nat."
+      }
+  , ErrorItem
+      { eiLabel = "internal_liquidity_below_zero_err"
+      , eiCode = 208
+      , eiDesc = "Liquidity went blow zero."
+      }
+  , ErrorItem
+      { eiLabel = "internal_209"
+      , eiCode = 209
+      , eiDesc = "Throw when `(p.dx - r.dx)` is not nat"
+      }
+  , ErrorItem
+      { eiLabel = "internal_insufficient_balance_err"
+      , eiCode = 210
+      , eiDesc = "Insufficient balance to trade asset `x` or `y`."
```

Comment:
Couple of things:

1. This can also happen when swapping `y` for `x`
2. We should make it clear that, in this instance, "balance" refers to the contract's balance and not the user's


How about:

```suggestion
      , eiDesc = "Contract does not have enough liquidity to execute the swap"
```

Reply:
Make sense, updated this.